crates/aggchain-proof-core/src/bridge/mod.rs

```diff
@@ -403,15 +422,21 @@ fn validate_hash_chain(
     Ok(())
 }
 
-fn filter_values<I: Eq + Hash + Copy>(removed: &[I], values: &[I]) -> Vec<I> {
+fn filter_values<I: Eq + Hash + Copy>(
+    removed: &[I],
+    values: &[I],
+) -> Result<Vec<I>, BridgeConstraintsError> {
```

Comment:
Could this be generalized a bit to take a key projection closure so it can be used for the bridge exits too?

A sketch:
```suggestion
fn filter_values<K: Eq + Copy, I: Hash + Clone>(
    removed: &[K],
    values: &[V],
    key_fn: impl FnMut(&V) -> K,
) -> Result<Vec<V>, BridgeConstraintsError> {
```